pyproj/_datadir.pyx

```diff
@@ -42,6 +41,7 @@ cdef class ContextManager:
             for iii in range(len(data_dir_list)):
                 b_data_dir = cstrencode(data_dir_list[iii])
                 c_data_dir[iii] = b_data_dir
+            proj_context_set_search_paths(NULL, len(data_dir_list), c_data_dir)
```

Comment:
Small remark here: it might be useful to add a comment on why this line is here (maybe linking to one of the issue explaining it). 

I was actually reading this code in one of the other PRs and already wondering why the search paths were set twice.

Reply:
That's a good idea. I would like to change this solution in 2.4 as I don't want to reset the global context each time. It's okay in this context as it is only set once.